br/pkg/restore/internal/prealloc_table_id/alloc.go

```diff
@@ -105,24 +173,35 @@ func (p *PreallocIDs) Alloc(m Allocator) error {
 	if p.reusableBorder <= p.start {
 		p.reusableBorder = p.start
 	}
-	idRange := p.reusableBorder - p.start + p.count
+
+	for i := p.start; i < p.reusableBorder; i++ {
+		p.allocRule[i] = i
```

Comment:
It may add too many non-existent IDs. If the difference between `p.start` and `p.reusableBorder` is too large.